crates/bevy_ui/src/update.rs

```diff
@@ -92,3 +92,55 @@ fn update_clipping(
         }
     }
 }
+
+pub fn update_ui_camera_system(
+    mut commands: Commands,
+    root_node_query: Query<
+        (Entity, Option<&UiCamera>),
+        (With<Node>, Without<Parent>, Changed<UiCamera>),
+    >,
+    changed_children_query: Query<(Entity, &UiCamera), Changed<Children>>,
+    children_query: Query<&Children>,
+    node_query: Query<Option<&UiCamera>, With<Node>>,
+) {
+    for (root_node, camera) in &root_node_query {
+        update_ui_camera(
+            &mut commands,
+            &children_query,
+            &node_query,
+            root_node,
+            camera,
+        );
+    }
+
+    for (parent, camera) in &changed_children_query {
+        // Recursively update all children
+        update_ui_camera(
+            &mut commands,
+            &children_query,
+            &node_query,
+            parent,
+            Some(camera),
+        );
+    }
+}
+
+fn update_ui_camera(
+    commands: &mut Commands,
+    children_query: &Query<&Children>,
+    node_query: &Query<Option<&UiCamera>, With<Node>>,
+    entity: Entity,
+    maybe_inherited_camera: Option<&UiCamera>,
+) {
+    let current_ui_camera = node_query.get(entity).unwrap();
+    let Some(ui_camera) = maybe_inherited_camera.or(current_ui_camera) else {
+        return;
+    };
+
+    if let Ok(children) = children_query.get(entity) {
+        for &child in children {
+            commands.entity(child).insert(ui_camera.clone());
```

Comment:
Could you only update this if it's needed, so that it doesn't trigger change detection?

Reply:
Resolved by https://github.com/bevyengine/bevy/pull/10559/commits/0b45e1e8c07337dd441def4b02a5bf0388ae7949

I now keep track of the change nodes to skip updating anything twice